src/Compilers/CSharp/Portable/Binder/Binder_Statements.cs

```diff
@@ -3115,14 +3114,22 @@ protected bool IsInAsyncMethod()
 
         protected bool IsTaskReturningAsyncMethod()
```

Comment:
I preferred my naming convention because it was less subtle :) even at the cost of wordiness
`IsNongenericTaskOrTasklikeReturningAsyncMethod`
`IsGenericTaskOrTasklikeReturningAsyncMethod`